### PR TITLE
refactor: resident save

### PR DESCRIPTION
**Pull request by shreyaskhamkar**
resident save and create resident units table

### CODE REVIEW

File: src/Core/Dhanman.MyHome.Application/Features/Residents/Queries/GetAllResidentsQueryHandler.cs [hunk 1]
```diff
@@ -31,9 +31,9 @@ public async Task<Result<ResidentListResponse>> Handle(GetAllResidentsQuery requ
               .AsNoTracking()                  
               .Select(e => new ResidentResponse(
                       e.Id,   
-                      e.UnitId,
+                      1,
```
Comment (abhikhot0448) [inline src/Core/Dhanman.MyHome.Application/Features/Residents/Queries/GetAllResidentsQueryHandler.cs:34]: Remove 1

File: src/Core/Dhanman.MyHome.Application/Features/Units/Queries/GetAllUnitsQueryHandler.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public async Task<Result<UnitListResponse>> Handle(GetAllUnitsQuery request, Can
                   .Select(e => new
                   {   Unit = e,
                       NumberOfMembers = _dbContext.SetInt<Resident>()
-                                         .Where(r => r.UnitId == e.Id)
+                                         .Where(r => 1 == e.Id)
```
Comment (abhikhot0448) [inline src/Core/Dhanman.MyHome.Application/Features/Units/Queries/GetAllUnitsQueryHandler.cs:38]: remove 1

File: src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/ResidentId.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dhanman.MyHome.Domain.Entities.ResidentUnits
+{
+    internal class ResidentId
```
Comment (abhikhot0448) [inline src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/ResidentId.cs:9]: make it record

File: src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/UnitId.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dhanman.MyHome.Domain.Entities.ResidentUnits
+{
+    internal class UnitId
+    {
```
Comment (abhikhot0448) [inline src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/UnitId.cs:10]: make it reord

**Review by abhikhot0448 — COMMENTED** [review]
make changes according to comment

File: src/Core/Dhanman.MyHome.Application/Features/Residents/Queries/GetAllResidentsQueryHandler.cs (hunk 1 above)
Comment (shreyaskhamkar) [reply src/Core/Dhanman.MyHome.Application/Features/Residents/Queries/GetAllResidentsQueryHandler.cs:34]: done

File: src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/ResidentId.cs (hunk 3 above)
Comment (shreyaskhamkar) [reply src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/ResidentId.cs:9]: done

File: src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/UnitId.cs (hunk 4 above)
Comment (shreyaskhamkar) [reply src/Core/Dhanman.MyHome.Domain/Entities/ResidentUnits/UnitId.cs:10]: done

**Pull request merged**